flow/layers/backdrop_filter_layer.cc

```diff
@@ -11,10 +11,34 @@ BackdropFilterLayer::BackdropFilterLayer(sk_sp<SkImageFilter> filter)
 
 BackdropFilterLayer::~BackdropFilterLayer() = default;
 
+void BackdropFilterLayer::initRetained(std::shared_ptr<Layer> retainedLayer) {
+  BackdropFilterLayer* bdfLayer = (BackdropFilterLayer*)retainedLayer.get();
+  if (bdfLayer->retainedBackdrop_) {
+    retainedBackdrop_ = bdfLayer->retainedBackdrop_;
+  } else {
+    retainedBackdrop_ = std::move(retainedLayer);
+  }
+}
+
 void BackdropFilterLayer::Paint(PaintContext& context) const {
   TRACE_EVENT0("flutter", "BackdropFilterLayer::Paint");
   FML_DCHECK(needs_painting());
 
+  if (retainedBackdrop_ && context.raster_cache && context.surface &&
+      context.view_embedder == nullptr) {
+    SkCanvas* canvas = context.leaf_nodes_canvas;
+    RasterCacheResult my_cache = context.raster_cache->GetSnapshot(
+        retainedBackdrop_.get(), context.gr_context, canvas->getTotalMatrix(),
+        context.surface, filter_.get());
+    if (my_cache.is_valid()) {
+      my_cache.drawSnapshot(canvas);
+      SkAutoCanvasRestore auto_restore(canvas, true);
```

Comment:
I don't see that code in PaintChildren, all I see is it iterating over the children and calling their Paint() (conditionally).

Reply:
Right, because `PaintChildren` itself doesn't change clip/matrix. But if a child's `Paint` would change them, the child will be responsible for restoring them.